test/ContractTestContext.ts

```diff
@@ -90,107 +86,101 @@ export type ContractTestContext = ArchContracts & {
 
-    await context.lvUSD.approve(context.exchanger.address, ethers.constants.MaxUint256);
-    await context.lvUSD.approve(context.coordinator.address, ethers.constants.MaxUint256);
+    /* expected contract roles when deployed to mainnet: */
+    const contractRoles = {
+        positionToken: {
+            executive: context.leverageEngine.address,
+        },
+    };
 
-    // Post init contracts
-    await Promise.all([
-        context.leverageEngine.init(
+    const initArgs = {
+        leverageEngine: [
             context.coordinator.address,
             context.positionToken.address,
             context.parameterStore.address,
             context.leverageAllocator.address,
             context.externalOUSD.address,
-        ),
-
-        context.coordinator.init(
-            context.lvUSD.address,
-            context.vault.address,
-            context.cdp.address,
+        ],
+        coordinator: [
+            context.lvUSDToken.address,
+            context.vaultOUSD.address,
+            context.cDPosition.address,
             context.externalOUSD.address,
             context.exchanger.address,
             context.parameterStore.address,
-        ),
-
-        context.exchanger.init(
+        ],
+        exchanger: [
             context.parameterStore.address,
             context.coordinator.address,
-            context.lvUSD.address,
+            context.lvUSDToken.address,
             context.externalOUSD.address,
             context.external3CRV.address,
             context.curveLvUSDPool.address,
             addressCurveOUSDPool,
-        ),
+        ],
+        vaultOUSD: [context.parameterStore.address, context.externalOUSD.address],
+        parameterStore: [context.treasurySigner.address],
+    };
 
-        context.vault.init(context.parameterStore.address, context.externalOUSD.address),
-        context.parameterStore.init(context.treasurySigner.address),
-        context.positionToken.init(),
-    ]);
+    /* call setRoles on all contracts, allowing any specified overrides from arguments: */
+    await Promise.all(Object.entries(contracts).map(async ([contractKey, contract]) => {
+        const roles = {
+            /* these defaults are temporary while  */
+            executive: context.owner.address,
+            governor: context.owner.address,
+            guardian: context.owner.address,
+            ...contractRoles[contractKey],
+            /* contractRoleOverrides allow tests to pass in an address override to make role based testing more clear */
+            ...contractRoleOverrides[contractKey] as Partial<DefaultRoles>,
+        };
+        await contract.connect(signerAdmin).setRoles(roles.executive, roles.governor, roles.guardian);
+        const args = initArgs[contractKey] || [];
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        await contract.connect(signerAdmin).init(...args);
```

Comment:
Both init and setRoles now happen consistently on all contracts. All contracts are expected to have these methods. Args for init added in the initArgs object above